yarn-project/types/src/logs/log_filter.ts

```diff
@@ -0,0 +1,48 @@
+import { AztecAddress, FunctionSelector } from '@aztec/circuits.js';
+
+import { LogId, TxHash } from '../index.js';
+
+/**
+ * Log filter used to fetch L2 logs.
+ */
+export type LogFilter = {
+  /**
+   * Hash of a transaction from which to fetch the logs.
+   * @remarks If this is set, `fromBlock` and `toBlock` can't be defined.
+   */
+  txHash?: TxHash;
+  /**
+   * The block number from which to start fetching logs (inclusive).
+   * @remarks If this is set, `txHash` and `fromLog` can't be defined.
+   */
+  fromBlock?: number;
+  /** The block number until which to fetch logs (not inclusive). */
+  toBlock?: number;
+  /**
+   * Log id after which to start fetching logs .
+   * @remarks If this is set, `fromBlock` and `txHash` can't be defined.
```

Comment:
I'd push back on this. The token for getting the next page should be compatible with any filter settings to be easy to use. A dev should be able to create any filter, send a request, and if they hit the max size they just add the `afterLog` to their filter for the next request, without needing any further changes.

In other words, I'd change it so `afterLog` overrides `fromBlock` (and others) rather than throwing.

Reply:
Would you make it override `txHash` as well?

I personally prefer a more strict behavior and not allowing devs to have conflicting filter params over having different priorities of params but I don't really have a strong opinion on this.

Reply:
I did the changes and now `afterLog` takes precedence over txHash, fromBlock and toBlock (all of them are ignored if afterLog is set).

But I realized that ignoring toBlock is probably not the right thing to do because it could happen that a user sets fromBlock and toBlock to span a range containing more than `maxLogs` logs and then he would unexpectedly receive logs from after toBlock. Would everyone agree that it makes sense to not ignore toBlock in such a situation? I would ignore toBlock only if afterLog.blockNumber >= toBlock.

Reply:
Agree, `toBlock` should not be ignored. And I would not ignore it even in the case where `afterLog.blockNumber >= toBlock`. The mental model for this should be: filters (such as `toBlock`) define a resultset, and `afterLog` indicates which "page" of the resultset gets returned to the user. So if the `afterLog` continuation parameter points to something outside the resultset, nothing should be returned.